contracts/script/LightClient.s.sol

```diff
@@ -1,48 +1,51 @@
 // SPDX-License-Identifier: UNLICENSED
-// This contract is auto-generated by gen-demo-contract. Direct edits are not recommended.
 pragma solidity ^0.8.0;
 
 import "forge-std/Script.sol";
-import { BN254 } from "bn254/BN254.sol";
+
 import { LightClient as LC } from "../src/LightClient.sol";
+import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+contract DeployLightClientContractScript is Script {
+    /// @notice deploys the impl, proxy & initializes the impl
+    /// @return proxyAddress The address of the proxy
+    /// @return admin The address of the admin
 
-contract DeployLightClientScript is Script {
-    function run() external {
+    function run(uint32 numBlocksPerEpoch, uint32 numInitValidators)
+        external
+        returns (address payable proxyAddress, address admin, LC.LightClientState memory)
+    {
         string memory seedPhrase = vm.envString("MNEMONIC");
-        uint256 privateKey = vm.deriveKey(seedPhrase, 0);
-        vm.startBroadcast(privateKey);
-
-        // For this version there will be only one epoch
-        uint32 blocksPerEpoch = type(uint32).max;
-
-        uint64 viewNum = 0;
-        uint64 blockHeight = 0;
-        BN254.ScalarField blockCommRoot = BN254.ScalarField.wrap(0);
-        BN254.ScalarField feeLedgerComm = BN254.ScalarField.wrap(0);
-        BN254.ScalarField stakeTableBlsKeyComm = BN254.ScalarField.wrap(
-            21279528985922333501911071464328386841445872547720180015435547221260402565727
-        );
-        BN254.ScalarField stakeTableSchnorrKeyComm = BN254.ScalarField.wrap(
-            7499525782104864821748149693309973670683978566054600186296438202262387687594
-        );
-        BN254.ScalarField stakeTableAmountComm = BN254.ScalarField.wrap(
-            17674948780218579510668373650995978278726151659230619414891977565353113435659
-        );
-        uint256 threshold = 3;
-
-        LC.LightClientState memory genesis = LC.LightClientState(
-            viewNum,
-            blockHeight,
-            blockCommRoot,
-            feeLedgerComm,
-            stakeTableBlsKeyComm,
-            stakeTableSchnorrKeyComm,
-            stakeTableAmountComm,
-            threshold
+        (admin,) = deriveRememberKey(seedPhrase, 0);
+        vm.startBroadcast(admin);
+
+        // TODO for a production deployment provide the right genesis state and value
+
+        string[] memory cmds = new string[](4);
+        cmds[0] = "diff-test";
+        cmds[1] = "mock-genesis";
+        cmds[2] = vm.toString(numBlocksPerEpoch);
```

Comment:
I think if we want to give an example, we should pass in these values?
ideally either via `.envrc` or inside `justfile`, or somehow read and parse from an artifact file?

the whole point of `gen-demo-contract.rs` was to populate these values/genesis artifacts directly embedding into the deploy script contract.

Reply:
With the current `LightClient.s.sol` script we do not need to run `gen-demo-contract.rs` which simplifies the development workflow and is also more robust I think. Also this approach avoids having many scripts for deploying the light client contract which is the main purpose of this PR.

Reply:
wait, so in production, how do you imagine the workflow works? how do we compute the genesis block? where is that information being stored and how do we load that into the contract constructor call arguments?

Reply:
We need to decide how to handle the deployment of contracts to production environment. This is also related to how we will upgrade contracts.

The way I see it is that the same script should be used for dev/staging/production environment with some flags. Based on the flags the script will call the right rust executable that will generate the correct initialization of the contract.   I am open to other ideas.

cc @sveitser @alysiahuggins

Reply:
Are we planning to deploy with a multi-sig wallet at this stage?

Reply:
> Based on the flags the script will call the rust executable that will generate the correct initialization of the contract

i see, so Chengyu's `gen-demo-contract.rs` assumes: "Rust produced init/genesis state/info --> encode into deployment script" workflow,

your code is "General deployment script --> call Rust code --> return init/genesis to deploy.sol --> encode info directly in solidity" workflow. 

I'm fine with this. Personally, I don't see clear advantage of either.

Reply:
I much prefer not having multiple scripts that do similar things and avoid code generation where feasible. I think if we want to decouple the creation of the genesis block and contract deployment we could save the genesis block to a JSON file, persist that JSON file and load it inside deployment scripts with https://book.getfoundry.sh/cheatcodes/parse-json. Not suggesting we do this in this PR though.

Reply:
>I much prefer not having multiple scripts that do similar things and avoid code generation where feasible.

make sense. 

imo, the JSON file approach is better for production, because we will have some static artifacts to version control and refer to, which is more portable and language-agnostic
the direct calling using forge FFI is better for testing since there's no point saving some temporary json file just for tests. 

good!